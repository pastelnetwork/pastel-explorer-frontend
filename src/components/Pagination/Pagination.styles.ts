import styled from 'styled-components/macro';

export const Wrapper = styled.div`
  display: inline-block;

  ul {
    display: flex;
    margin: 0;
    padding: 0;
    list-style: none;

    li {
      margin: 0 5px;

      a {
        display: block;
        padding: 5px 10px;
        background: ${props => props.theme.filter.background};
        border-radius: 4px;
        border: 0;
        text-align: center;
        cursor: pointer;
        text-decoration: none;
        transition: all 0.2s ease;
        color: ${props => props.theme.palette.text.primary};
      }

      &.selected,
      &:hover {
        a {
          background: ${props => props.theme.sidebar.menu.toggle.switch};
          color: #fff;
        }
      }

      &.selected {
        a {
          cursor: default;
        }
      }

      &.disabled,
      &.disabled:hover {
        a {
          cursor: default;
          background: ${props => props.theme.table.odd};
          color: ${props => props.theme.palette.text.primary};
        }
      }

      &.break,
      &.break:hover {
        a {
          background: transparent;
          color: ${props => props.theme.palette.text.primary};
        }
      }
    }
  }

  ${props => props.theme.breakpoints.down(600)} {
    ul {
      flex-wrap: wrap;
      justify-content: center;

      li {
        margin-bottom: 10px;
      }
    }
  }
`;
