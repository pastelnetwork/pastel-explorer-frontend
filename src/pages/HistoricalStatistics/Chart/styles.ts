import { makeStyles } from '@material-ui/core/styles';
import { TAppTheme } from '@theme/index';

export const eChartLineStyles = makeStyles((theme: TAppTheme) => ({
  container: {
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    padding: '0 20px',
    textAlign: 'center',
    boxSizing: 'border-box',
    position: 'relative',
    color: 'rgb(171, 170, 193)',
    [theme.breakpoints.down('sm')]: {
      padding: 0,
    },
  },
  lineChartWrap: {
    width: '100%',
    height: 'auto',
    marginBottom: 20,
  },
  lineChartContainer: {
    display: 'block',
    width: '100%',
    height: '100%',
    backgroundColor: '#212121',
  },
  lineChartTitle: {
    width: '100%',
    textAlign: 'left',
    fontSize: '30px',
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
  },
  lineChartHeader: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
    paddingTop: '30px',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      paddingTop: 10,
      paddingLeft: 10,
      justifyContent: 'flex-start',
      flexDirection: 'column',
    },
  },
  uploadButton: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: '#6151d0',
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '14px',
    padding: '8px 16px',
    borderRadius: '4px',
    marginLeft: '8px',
    marginRight: '8px',
    '&:hover': {
      cursor: 'pointer',
      opacity: 0.8,
    },
  },
  reactECharts: {
    width: '100%',
    height: 'calc(100vh - 450px) !important',
  },
  granularitySelect: {
    display: 'flex',
    width: '100%',
    maxWidth: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15,
    marginRight: 50,
  },
  periodSelect: {
    display: 'flex',
    minWidth: '200px',
    // justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: '15px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 5,
    },
  },
  lineChartDownloadButtonBar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    height: '30px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 10,
      justifyContent: 'flex-start',
    },
  },
  lineChartFooter: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingBottom: 30,
    [theme.breakpoints.down('sm')]: {
      flexWrap: 'wrap',
      justifyContent: 'flex-start',
      paddingBottom: 10,
    },
  },
  lineChartThemeSelect: {
    display: 'flex',
    minWidth: '150px',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  themeSelectButton: {
    width: '30px',
    height: '30px',
    borderRadius: '5px',
  },
  filterButton: {
    backgroundColor: '#eee',
    color: '#212124',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '12px',
    padding: '4px 8px',
    borderRadius: '3px',
    borderWidth: '1px',
    borderColor: '#212124',
    marginLeft: '4px',
    marginRight: '4px',
    '&:hover': {
      cursor: 'pointer',
      opacity: '0.8',
    },
  },
  activeButton: {
    backgroundColor: '#c3921f',
    borderColor: '#c3921f',
    color: '#fff',
  },
  activeThemeButton: {
    borderColor: '#008000',
    borderWidth: '5px',
  },
}));
