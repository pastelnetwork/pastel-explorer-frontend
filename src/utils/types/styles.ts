/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/ban-types */
import { SpacingProps } from '@material-ui/system';
import { Theme } from '@material-ui/core';

export type GlobalStyleProps = {
  theme: Theme & { palette: any };
};

export interface MuiButtonSpacingType extends SpacingProps {
  component?: React.PropsWithoutRef<{}>;
  to?: string;
}
