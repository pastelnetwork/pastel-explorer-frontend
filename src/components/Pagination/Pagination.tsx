import ReactPaginate from 'react-paginate';

import * as Styles from './Pagination.styles';

interface IPaginationProps {
  totalPage: number;
  pageRangeDisplayed?: number;
  nextLabel?: React.ReactNode;
  previousLabel?: React.ReactNode;
  breakLabel?: React.ReactNode;
  onPageChange?: (_page: number) => void;
  defaultPage?: number;
}

const Pagination: React.FC<IPaginationProps> = ({
  totalPage,
  pageRangeDisplayed = 3,
  nextLabel = '>',
  previousLabel = '<',
  breakLabel = '...',
  onPageChange,
  defaultPage = 0,
}) => {
  const handlePageClick = (event: { selected: number }) => {
    if (onPageChange) {
      onPageChange(event.selected);
    }
  };
  return (
    <Styles.Wrapper>
      <ReactPaginate
        breakLabel={breakLabel}
        nextLabel={nextLabel}
        previousLabel={previousLabel}
        onPageChange={handlePageClick}
        pageRangeDisplayed={pageRangeDisplayed}
        pageCount={totalPage}
        forcePage={defaultPage}
      />
    </Styles.Wrapper>
  );
};

export default Pagination;
