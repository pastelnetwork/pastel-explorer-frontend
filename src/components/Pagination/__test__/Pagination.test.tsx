import { shallow } from 'enzyme';
import ReactPaginate from 'react-paginate';

import 'jest-styled-components';

import * as Styles from '../Pagination.styles';
import Pagination from '../Pagination';

describe('components/Pagination', () => {
  const wrapper = shallow(<Pagination totalPage={5} />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <ReactPaginate>', () => {
    expect(wrapper.find(ReactPaginate).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.Wrapper>', () => {
    expect(wrapper.find(Styles.Wrapper).length).toBeGreaterThanOrEqual(1);
  });
});
