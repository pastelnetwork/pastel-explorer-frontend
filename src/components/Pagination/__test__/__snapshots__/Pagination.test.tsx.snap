// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Pagination renders correctly 1`] = `ShallowWrapper {}`;
